pkg/psmdb/backup/pbm.go

```diff
@@ -146,3 +146,52 @@ func secret(cl client.Client, namespace, secretName string) (*corev1.Secret, err
 	err := cl.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: namespace}, secret)
 	return secret, err
 }
+
+type LockHeaderPredicate func(pbm.LockHeader) bool
+
+func NotPITRLock(l pbm.LockHeader) bool {
+	return l.Type != pbm.CmdPITR
+}
+
+func NotJobLock(j Job) LockHeaderPredicate {
+	return func(h pbm.LockHeader) bool {
+		var jobCommand pbm.Command
+
+		switch j.Type {
+		case TypeBackup:
+			jobCommand = pbm.CmdBackup
+		case TypeRestore:
+			jobCommand = pbm.CmdRestore
+		default:
+			return true
+		}
+
+		return h.Type != jobCommand
+	}
+}
+
+func (b *PBM) HasLocks(predicates ...LockHeaderPredicate) (bool, error) {
+	locks, err := b.C.GetLocks(&pbm.LockHeader{})
+	if err != nil {
+		return false, errors.Wrap(err, "getting lock data")
+	}
+
+	allowedByAll := func(l pbm.LockHeader) bool {
+		for _, allow := range predicates {
+			if !allow(l) {
+				return false
+			}
+		}
+		return true
+	}
+
+	for _, l := range locks {
+		if !allowedByAll(l.LockHeader) {
+			continue
```

Comment:
can we make this part is a little bit more readable as a loop above?
Like
```
for _,l := range locks{
  if allowedByAll(l.LockHeader){
    return true,nil
  }
}
return false, nil
```

Reply:
Yes! Also, I think it should be renamed to `allowedByAllPredicates` cause it's not clear from the function call.